src/vs/workbench/contrib/scm/browser/scmViewPane.ts

```diff
@@ -1182,6 +1182,20 @@ class ViewModel {
 			if (e.reason === WillSaveStateReason.SHUTDOWN) {
```

Comment:
Why do we only store on shutdown, dont we have to store in all cases now?

Reply:
`scm.viewState` isn't portable in its current form since it references stringified URIs: 
https://github.com/microsoft/vscode/blob/76d228b89b3324c390c798e84557113c0468f6d9/src/vs/workbench/contrib/scm/browser/scmViewPane.ts#L786-L807

So I do not want to try to adopt workspace state for it just yet, I think we should try to tackle this as part of a broader adoption effort.

Reply:
Oh ok. I thought that was the idea of this PR but now I see how there is other state you roam.